packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -145,17 +207,111 @@ class _PopupMenuRoute<T> extends PopupRoute<T> {
   Color get barrierColor => null;
 
   Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> forwardAnimation) {
-    return new _PopupMenu(route: this);
+    int selectedIndex = -1;
+    if (initialValue != null) {
+      for (int i = 0; i < items.length; i++)
+        if (initialValue == items[i].value) {
+          selectedIndex = i;
+          break;
+        }
+    }
+    final Size screenSize = MediaQuery.of(context).size;
+    return new SizedBox(
+      width: screenSize.width,
+      height: screenSize.height,
+      child: new CustomOneChildLayout(
+        delegate: new _PopupMenuRouteLayout(position, selectedIndex),
+        child: new _PopupMenu(route: this)
+      )
+    );
   }
 }
 
-Future showMenu({ BuildContext context, ModalPosition position, List<PopupMenuItem> items, int elevation: 8 }) {
+/// Show a popup menu that contains the [items] at [position]. If [initialValue]
+/// is specified then the first item with a matching value will be highlighted
+/// and the value of [position] implies where the left, center point of the
+/// highlighted item should appear. If [initialValue] is not specified then position
+/// implies the menu's origin.
+Future showMenu({
```

Comment:
you can use the new syntax for generic methods:
`Future/*<T>*/ showMenu/*<T>*/({`